src/Http/Controllers/PageController.php

```diff
@@ -13,24 +13,11 @@ class PageController extends BaseController
 {
     public function __invoke(Request $request, Page $page, ?Page $childPage = null, Page $grandChildPage = null)
     {
-        // If location based grand child is only child, redirect!!
-        if ($grandChildPage) {
-            if ($grandChildPage->location_based && $grandChildPage->is_only_child) {
-                return redirect(url($grandChildPage->path));
-            }
-        } elseif ($childPage) {
-            // If location based child is only child, redirect!!!
-            if ($childPage->location_based && $childPage->is_only_child) {
-                return redirect(url($childPage->path));
-            }
-        } else {
-            // If there is only  location based root page, redirect home!
-            if ($page->is_only_root_location) {
-                return redirect(url($page->path));
-            }
+        $leafPage = $grandChildPage ?: ($childPage ?: $page);
```

Comment:
I see this logic in a few spots. It's a bit hard to parse the compound ternary and sooner or later a dev is going to forget those parenthesis which will alter the behavior.

Wonder if wrapping in a simple helper would be... helpful: `determine_leaf_page($grandChildPage, $childPage, $page)`